### PR TITLE
Issue #7742: Update AbstractChecks to log DetailAST - LambdaParameter…

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Issue #7742 is fixed.
[Diff report](https://abhishek-kumar09.github.io/7742/reports/diff/index.html)

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
No change in diff report, maybe due to this fix require only xpathregression tests, and no change in `LambdaParameterNameCheck.java` is done.

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Test
+    public void testTwo() throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionLambdaParameterName2.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(LambdaParameterNameCheck.class);
+        moduleConfig.addAttribute("format", "^[_][a-zA-Z0-9]*$");
+        final String nonDefaultPattern = "^[_][a-zA-Z0-9]*$";
```
Comment (strkkk) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java]: Please use this variable in statement before

Comment (strkkk) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java]: first group in regex is redundant, can be simplified to just underscore

Comment (Abhishek-kumar09) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java]: OK

Comment (Abhishek-kumar09) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java]: Done.

**Comment by rnveach** [comment]
@Abhishek-kumar09 Appveyor failure looks correct.

**Comment by Abhishek-kumar09** [comment]
I have updated the changes.

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java (hunk 1 above)
Comment (strkkk) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java]: @Abhishek-kumar09 there is a misunderstanding. By first group I meant `[_]`, this group can be simplified to `_`, no reason to delete rest of regex.
Please correct to `^_[a-zA-Z0-9]*$`

**Comment by strkkk** [comment]
@Abhishek-kumar09 Please see https://github.com/checkstyle/checkstyle/pull/7949#discussion_r397657318

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java (hunk 1 above)
Comment (Abhishek-kumar09) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionLambdaParameterNameTest.java]: Sorry, I misunderstood that.

File: src/it/resources/org/checkstyle/suppressionxpathfilter/lambdaparametername/SuppressionXpathRegressionLambdaParameterName2.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.checkstyle.suppressionxpathfilter.lambdaparametername;
+
+import java.util.function.Function;
+
+public class SuppressionXpathRegressionLambdaParameterName2 {
+    void test() {
+        Function<String, String> trimmer = s -> s.trim(); // warn
```
Comment (rnveach) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/lambdaparametername/SuppressionXpathRegressionLambdaParameterName2.java]: Inputs between the 2 should be more unique.

Please add a parenthesis around the lambda parameter like https://github.com/checkstyle/checkstyle/blob/3835575706d8240b3ec054d17ed48f692b168611/src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/lambdaparametername/InputLambdaParameterName.java#L10 /

Comment (Abhishek-kumar09) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/lambdaparametername/SuppressionXpathRegressionLambdaParameterName2.java]: OK,

**Pull request merged**